baseband/mark4/tests/test_mark4.py

```diff
@@ -386,6 +362,89 @@ def test_frame(self, tmpdir):
                                    ref_time=Time('2019:1:9:00:00'))
             assert frame7 == frame
 
+    def test_frame_getitem_setitem(self):
+        with mark4.open(SAMPLE_FILE, 'rb') as fh:
+            fh.seek(0xa88)
+            frame = fh.read_frame(ntrack=64,
+                                  ref_time=Time('2009:345:15:00:00'))
+            header = frame.header
+            data = frame.data
+        # Check __getitem__.
+        assert np.all(frame['magnitude_bit'] == header['magnitude_bit'])
+        # Only invalid values (compare with data to ensure shape is right).
+        assert np.all(frame[10:90] == data[10:90])
+        assert np.all(frame[10:90:5] == data[10:90:5])
+        assert np.all(frame[10:90:5, :4] == data[10:90:5, :4])
+        # Mixed invalid, valid.
+        assert np.all(frame[635:655] == data[635:655])
+        # Check we do stepping correctly around the boundary.
+        for start in range(634, 642):
+            assert np.all(frame[start:655:5] == data[start:655:5])
+        assert np.all(frame[635:655:5, 5] == data[635:655:5, 5])
+        # All valid.
+        assert np.all(frame[935:955] == data[935:955])
+        assert np.all(frame[935:955:5] == data[935:955:5])
+        assert np.all(frame[935:955:5, 5] == data[935:955:5, 5])
+        # Check scalar, including whether the shape is correct.
+        assert frame[935].shape == data[935].shape
+        assert np.all(frame[639] == data[639])
+        assert np.all(frame[640] == data[640])
+        # Check __setitem__.
+        # Default frame read from a file is not mutable.
+        with pytest.raises(ValueError):
+            frame[640:] = 0.
+        # So create a mutable version of the frame.
+        frame = mark4.Mark4Frame.fromdata(frame.data, header.copy())
+        # Check that setting invalid part has no effect.
+        frame[10:90] == 1.
+        frame[10:90:5] == 1.
+        frame[10:90:5, :4] == 1.
+        assert np.all(frame[:640] == 0.)
+        # Mixed invalid, valid.
+        frame[635:655] = 1.
+        assert np.all(frame[635:640] == 0.)
```

Comment:
Nitpicky (since the two are equal), but we technically should be checking the frame against `frame.invalid_data_value` instead of 0, as with (most of?) the stream reader tests.